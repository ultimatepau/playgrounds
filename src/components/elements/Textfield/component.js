import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import Typography from '@telkomdesign/tedis-reactjs-component/lib/Typography';

export default class Textfield extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      active: false,
      hover: false,
      inputSelection: false,
      value: this.props.meta ? this.props.meta.value : '',
      updateSelection: false
    };
    this.inputElRef = React.createRef();
  }

  componentDidUpdate() {
    if (this.state.updateSelection) {
      const selection = this.state.updateSelection;
      this.inputElRef.current.selectionStart = selection.start;
      this.inputElRef.current.selectionEnd = selection.end;
      this.setState({ updateSelection: false });
    }
  }

  _handlerClick(active) {
    this.setState({ active });
  }

  _handleChange(e) {
    const { input } = this.props;
    e.preventDefault();
    const { normalize, variant } = this.props.meta;
    let newValue;
    if (normalize) {
      const normalValue = normalize(e.target.value);
      newValue = normalValue !== undefined ? normalValue : input.value;
    } else {
      newValue = e.target.value;
    }
    this.setState({ value: newValue }, () => {
      input.onChange(this.state.value);

      if (variant !== 'area-filled' && variant !== 'area-line') {
        this.inputElRef.current.style.height = 'inherit';

        if ((this.inputElRef.current.value.match(/\n/g) || []).length === 0) {
          this.inputElRef.current.style.height = `15px`;
        } else {
          this.inputElRef.current.style.height = `${this.inputElRef.current.scrollHeight}px`;
        }
      }
      this.setState({ updateSelection: { ...this.state.inputSelection } });
    });
  }

  _handlerClickWrapper() {
    const { onClick } = this.props;

    this.inputElRef.current.focus();

    if (typeof onClick === 'function') {
      onClick();
    }
  }

  _handlerFocus() {
    const { input } = this.props;
    const { onFocus = () => { } } = input;

    this.saveCursorPosition();

    onFocus();
  }

  _handlerHover(hover) {
    this.setState({ hover });
  }

  _handlerKeydown(e) {
    if (e.key === 'Backspace') {
      this.saveCursorPosition(-1);
    } else {
      this.saveCursorPosition(1);
    }
  }

  _handlerKeyUp(e) {
    const { input } = this.props;
    const number = new RegExp(/^\d+$/);
    let increment;
    if (e.key === 'ArrowUp' && number.test(e.target.value)) {
      increment = 1;
      this.saveCursorPosition(2);
    }
    if (e.key === 'ArrowDown') {
      increment = -1;
    }
    if (increment) {
      this.setState({ value: parseInt(e.target.value) + increment }, () => {
        input.onChange(this.state.value);
      });
    }
  }

  saveCursorPosition(increment) {
    this.setState({
      inputSelection: {
        start: this.inputElRef.current.selectionStart + increment,
        end: this.inputElRef.current.selectionEnd + increment
      }
    });
  }

  render() {
    const { classes, input, meta: {
      'icon-align': iconAlign,
      className : classNameProps,
      color,
      disabled,
      error,
      icon: Icon,
      helper = '',
      placeholder = '',
      type,
      variant = 'basic-filled'
    }, onClickIcon } = this.props;
    const {
      onBlur = () => { },
      onMouseEnter = () => { },
      onMouseLeave = () => { }
    } = input;
    const { active, hover } = this.state;
    const variantData = variant ? variant : 'basic-filled';
    const Tag = variantData === 'basic-filled' || variantData === 'basic-line' ? 'input' : 'textarea';
    const inputForm = (
      <Tag
        className={classNames(
          classes.formInput,
          {
            [classes.formInputFilled]:
              variantData.match(/^[a-z]{0,10}-filled$/g) && !(this.state.hover || disabled),
            [classes.formInputFilledHover]:
              variantData.match(/^[a-z]{0,10}-filled$/g) && (this.state.hover || disabled),
            [classes.formInputLine]:
              variantData.match(/^[a-z]{0,10}-line$/g),
          },
          { [classes.textarea]: Tag === 'textarea' },
          { [classes.formInputDisabled]: disabled && variantData.match(/^[a-z]{0,10}-filled$/g) },
          { [classes.formInputError]: error },
          { [classes.variantLine]: variantData === 'area-line' || variantData === 'multiline-line' || variantData === 'basic-line' },
          { [classes.textareaHideScroll]: Tag === 'textarea' && (variantData === 'multiline-filled' || variantData === 'multiline-line') },
          { [classes.textAreaVariant]: Tag === 'textarea' && (variantData === 'area-filled' || variantData === 'area-line') },
        )}
        disabled={disabled}
        id={input.name}
        name={input.name}
        onBlur={onBlur.bind(this)}
        onChange={this._handleChange.bind(this)}
        onFocus={this._handlerFocus.bind(this)}
        onKeyDown={this._handlerKeydown.bind(this)}
        onKeyUp={this._handlerKeyUp.bind(this)}
        onMouseEnter={onMouseEnter.bind(this)}
        onMouseLeave={onMouseLeave.bind(this)}
        placeholder={placeholder}
        ref={this.inputElRef}
        style={{ color: color }}
        type={type}
        value={input.value}
      />);

    return (
      <section
        onClick={this._handlerClickWrapper.bind(this)}
        onMouseEnter={this._handlerHover.bind(this, true)}
        onMouseLeave={this._handlerHover.bind(this, false)}
      >
        <div
          className={classNames(
            classes.formInputWrap,
            { [classes.formActive]: active },
            {
              [classes.iconRight]: iconAlign === 'right',
              [classes.iconLeft]: iconAlign === 'left'
            },
            {
              [classes.formInputWrapFilled]:
                variantData.match(/^[a-z]{0,10}-filled$/g) && !(hover || disabled),
              [classes.formInputWrapFilledHover]:
                variantData.match(/^[a-z]{0,10}-filled$/g) && (hover || disabled),
              [classes.formInputWrapLine]:
                variantData.match(/^[a-z]{0,10}-line$/g),
            },
            {
              [classes.formInputWrapDisabled]: disabled &&
                variantData.match(/^[a-z]{0,10}-filled$/g)
            },
            { [classes.formInputWrapError]: error },
            { [classes.textareaWraper]: Tag === 'textarea' },
            classNameProps
          )}
        >
          {
            //showwing icon
            Icon && <div
              className={classNames(classes.iconWrap)}
              onClick={onClickIcon}
            >
              <Icon fontSize="inherit" />
            </div>
          }
          <div
            className={classNames(classes.formInputTextWrap, { [classes.formInputActive]: active })}
            id="wrapper"
            onBlur={this._handlerClick.bind(this, false)}
            onFocus={this._handlerClick.bind(this, true)}
          >
            {inputForm}
            <label
              className={classNames(
                classes.formInputTextLabel,
                { [classes.formLabelActive]: active },
                {
                  [classes.formInputDisabled]: disabled &&
                    variantData.match(/^[a-z]{0,10}-filled$/g)
                },
                { [classes.formInputTextLabelError]: error }
              )}
              htmlFor={input.name}
            >
              <b>
                <Typography variant="overline2">{placeholder}</Typography>
              </b>
            </label>
          </div>
        </div>
        <div className={classNames(classes.helperWrap, { [classes.helperWrapError]: error })}>
          <Typography variant="overline1">{helper}</Typography>
        </div>
      </section >
    );
  }
}

Textfield.propTypes = {
  classes: PropTypes.object.isRequired,
  input: PropTypes.object,
  meta: PropTypes.object,
  onClick: PropTypes.func,
  onClickIcon: PropTypes.func
};
Textfield.defaultProps = {
  input: {},
  meta: {
    'icon-align': 'right',
    color: 'primary',
    disabled: 'false',
    error: null,
    icon: null,
    helper: '',
    placeholder: '',
    type: null,
    variant: 'basic-filled'
  },
  onClick: () => { },
  onClickIcon: () => { },
};
