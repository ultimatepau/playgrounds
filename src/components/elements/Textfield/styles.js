import { Grey, Red } from '@telkomdesign/colors/lib';

const styles = theme => ({
  '@global': {
    input: {
      '&:-webkit-autofill': {
        WebkitBoxShadow: '0 0 0 30px white inset'
      },
      '&:disabled': {
        backgroundColor: 'inherit'
      }
    }
  },
  content: {
    content: '',
    display: 'block',
    paddingTop: '100%'
  },
  formActive: {
    borderBottom: 'solid ' + theme.palette.base.primary.main + ' !important',
    boxSizing: 'border-box',
    MozBoxSizing: 'border-box',
    WebkitBoxSizing: 'border-box'
  },
  formColorPreview: {
    border: 'solid ' + theme.palette.base.primary.main,
    borderRadius: '4px',
    borderWidth: '0px 0.5px 0px 0px',
    maxHeight: '100%',
    minHeight: '99%',
    position: 'relative',
    width: '50%'
  },
  formIcon: {
    margin: '0 -5px 0 5px',
    maxHeight: '40%',
    width: '10%'
  },
  formInput: {
    '&:-moz-placeholder': {
      color: 'transparent'
    },
    '&:-ms-input-placeholder': {
      color: 'transparent'
    },
    '&:-webkit-autofill': {
      color: 'grey',
      WebkitBoxShadow: '0 0 0 30px white inset'
    },
    '&::-moz-placeholder': {
      color: 'transparent'
    },
    '&::-webkit-input-placeholder': {
      color: 'transparent'
    },
    '&::placeholder': {
      color: 'transparent'
    },
    '&:focus': {
      outline: 'none'
    },
    '&:focus + label': {
      color: theme.palette.base.primary.dark,
      fontWeight: '600',
      transform: 'translate(0, 0) scale(1) !important',
      transformOrigin: 'left bottom'
    },
    '&:placeholder-shown + label': {
      cursor: 'text',
      maxWidth: '62%',
      overflow: 'hidden',
      textOverflow: 'ellipsis',
      transform: 'translate(0, 0.8rem) scale(1.4)',
      transformOrigin: 'left bottom',
      whiteSpace: 'nowrap'
    },
    border: '0px',
    color: Grey[90],
    fontFamily: theme.typography.fontFamily,
    fontSize: '14px',
    fontWeight: '500',
    width: '99%'
  },
  formInputColor: {
    width: '69% !important'
  },
  formInputDisabled: {
    color: `${Grey[50]} !important`,
    cursor: 'not-allowed !important'
  },
  formInputError: {
    '&:-webkit-autofill': {
      WebkitBoxShadow: `0 0 0 30px ${Red[20]} inset`
    },
    backgroundColor: `${Red[20]} !important`
  },
  formInputFilled: {
    backgroundColor: Grey[20]
  },
  formInputFilledHover: {
    backgroundColor: Grey[30]
  },
  formInputTextLabel: {
    color: Grey[90],
    fontSize: '10px',
    textTransform: 'capitalize',
    transition: 'all 0.2s'
  },
  formInputTextLabelError: {
    color: Red[60]
  },
  formInputTextWrap: {
    display: 'flex',
    flex: '1',
    flexDirection: 'column-reverse',
    maxWidth: '100%',
    padding: '15px 0px'
  },
  formInputWrap: {
    alignContent: 'center',
    alignItems: 'center',
    borderBottom: 'solid ' + Grey[70],
    borderRadius: '2px',
    boxSizing: 'border-box',
    display: 'flex',
    fontFamily: theme.typography.fontFamily,
    height: '50px',
    MozBoxSizing: 'border-box',
    padding: '0px 8px',
    WebkitBoxSizing: 'border-box',
    width: '100%'
  },
  formInputWrapDisabled: {
    borderBottom: 'solid ' + Grey[50],
    color: Grey[50],
    cursor: 'not-allowed'
  },
  formInputWrapError: {
    backgroundColor: `${Red[20]} !important`,
    borderBottom: 'solid ' + Red[60]
  },
  formInputWrapFilled: {
    backgroundColor: Grey[20]
  },
  formInputWrapFilledHover: {
    backgroundColor: Grey[30]
  },
  formLabelActive: {
    color: theme.palette.base.primary.dark,
    fontWeight: '600'
  },
  helperWrap: {
    position: 'absolute',
    color: Grey[90],
    margin: '4px 8px'
  },
  helperWrapError: {
    color: Red[60]
  },
  iconLeft: {
    flexDirection: 'row'
  },
  iconRight: {
    flexDirection: 'row-reverse'
  },
  iconWrap: {
    cursor: 'pointer',
    color: Grey[90],
    display: 'flex',
    fontSize: '20px',
    marginRight: '8px'
  },
  textarea: {
    height: '15px',
    resize: 'none'
  },
  textareaHideScroll: {
    '&::-webkit-scrollbar': {
      background: 'transparent !important',
      width: '0px !important'
    }
  },
  textAreaVariant: {
    height: '150px'
  },
  textareaWraper: {
    height: 'inherit'
  },
  variantLine: {
    backgroundColor: 'inherit'
  }
});

export default styles;
