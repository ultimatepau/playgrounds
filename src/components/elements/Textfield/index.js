import withStyles from 'react-jss';
import TextField from './component';
import styles from './styles';

export default withStyles(styles)(TextField);