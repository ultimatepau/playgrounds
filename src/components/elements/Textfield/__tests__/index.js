import React from 'react';
import themeMount from '../../utils/themeMount';
import Textfield from '../index';

jest.mock('react-router-dom');
jest.mock('../component');
jest.mock('../styles');

describe('Textfield', () => {
  it('renders correctly', () => {
    const [styleSheets, component] = themeMount(<Textfield />);
    expect(styleSheets.toString()).toMatchSnapshot();
    expect(component).toMatchSnapshot();
  });
});