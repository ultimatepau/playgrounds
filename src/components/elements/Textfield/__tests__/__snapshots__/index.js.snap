// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textfield renders correctly 1`] = `""`;

exports[`Textfield renders correctly 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): ShallowWrapper {
    Symbol(enzyme.__root__): [Circular],
    Symbol(enzyme.__unrendered__): <JssProvider
      registry={
        SheetsRegistry {
          "registry": Array [],
        }
      }
    >
      <ThemeProvider
        theme={
          Object {
            "palette": Object {
              "base": Object {
                "primary": Object {
                  "dark": "#BA0D0D",
                  "light": "#F55151",
                  "main": "#DE1B1B",
                },
                "secondary": Object {
                  "dark": "#7D3CBE",
                  "light": "#AD73E8",
                  "main": "#9657D6",
                },
                "tertiary": Object {
                  "dark": "#1D5BBF",
                  "light": "#5691F0",
                  "main": "#3071D9",
                },
              },
              "common": Object {
                "black": "#212121",
                "white": "#FFFFFF",
              },
              "event": Object {
                "disabled": Object {
                  "main": "#BDBDBD",
                },
                "error": Object {
                  "light": "#FAF0F0",
                  "main": "#DE1B1B",
                },
                "information": Object {
                  "light": "#EAF5FA",
                  "main": "#229BD8",
                },
                "success": Object {
                  "light": "#F4FCE3",
                  "main": "#74B816",
                },
                "warning": Object {
                  "light": "#FFF9DB",
                  "main": "#F59F00",
                },
              },
              "grey": Object {
                "10": "#FAFAFA",
                "100": "#212121",
                "20": "#F5F5F5",
                "30": "#EEEEEE",
                "40": "#E0E0E0",
                "50": "#BDBDBD",
                "60": "#9E9E9E",
                "70": "#757575",
                "80": "#616161",
                "90": "#424242",
              },
              "text": Object {
                "primary": Object {
                  "dark": "#F55151",
                  "light": "#900E0E",
                },
                "secondary": Object {
                  "dark": "#AD73E8",
                  "light": "#632B9C",
                },
                "tertiary": Object {
                  "dark": "#AD73E8",
                  "light": "#632B9C",
                },
              },
            },
            "pxToRem": [Function],
            "typography": Object {
              "body1": Object {
                "fontSize": "16px",
                "fontWeight": 400,
              },
              "body2": Object {
                "fontSize": "14px",
                "fontWeight": 400,
              },
              "button": Object {
                "fontSize": "14px",
                "fontWeight": 700,
              },
              "caption": Object {
                "fontSize": "12px",
                "fontWeight": 400,
              },
              "fontFamily": "Roboto",
              "fontSize": 14,
              "fontSizeInherit": "inherit",
              "fontWeightBold": 700,
              "fontWeightLight": 300,
              "fontWeightMedium": 500,
              "fontWeightRegular": 400,
              "h1": Object {
                "fontSize": "60px",
                "fontWeight": 300,
              },
              "h2": Object {
                "fontSize": "48px",
                "fontWeight": 400,
              },
              "h3": Object {
                "fontSize": "34px",
                "fontWeight": 400,
              },
              "h4": Object {
                "fontSize": "24px",
                "fontWeight": 400,
              },
              "h5": Object {
                "fontSize": "20px",
                "fontWeight": 500,
              },
              "overline1": Object {
                "fontSize": "10px",
                "fontWeight": 400,
              },
              "overline2": Object {
                "fontSize": "10px",
                "fontWeight": 500,
              },
              "subtitle1": Object {
                "fontSize": "16px",
                "fontWeight": 400,
              },
              "subtitle2": Object {
                "fontSize": "14px",
                "fontWeight": 500,
              },
            },
          }
        }
      >
        <Jss(Textfield)
          _renderid="rendered-component"
          input={Object {}}
          meta={
            Object {
              "color": "primary",
              "disabled": "false",
              "error": null,
              "helper": "",
              "icon": null,
              "icon-align": "right",
              "placeholder": "",
              "type": null,
              "variant": "basic-filled",
            }
          }
          onClick={[MockFunction]}
        />
      </ThemeProvider>
    </JssProvider>,
    Symbol(enzyme.__renderer__): Object {
      "batchedUpdates": [Function],
      "checkPropTypes": [Function],
      "getNode": [Function],
      "render": [Function],
      "simulateError": [Function],
      "simulateEvent": [Function],
      "unmount": [Function],
    },
    Symbol(enzyme.__node__): Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Jss(Textfield)
          _renderid="rendered-component"
          input={Object {}}
          meta={
            Object {
              "color": "primary",
              "disabled": "false",
              "error": null,
              "helper": "",
              "icon": null,
              "icon-align": "right",
              "placeholder": "",
              "type": null,
              "variant": "basic-filled",
            }
          }
          onClick={[MockFunction]}
        />,
        "theme": Object {
          "palette": Object {
            "base": Object {
              "primary": Object {
                "dark": "#BA0D0D",
                "light": "#F55151",
                "main": "#DE1B1B",
              },
              "secondary": Object {
                "dark": "#7D3CBE",
                "light": "#AD73E8",
                "main": "#9657D6",
              },
              "tertiary": Object {
                "dark": "#1D5BBF",
                "light": "#5691F0",
                "main": "#3071D9",
              },
            },
            "common": Object {
              "black": "#212121",
              "white": "#FFFFFF",
            },
            "event": Object {
              "disabled": Object {
                "main": "#BDBDBD",
              },
              "error": Object {
                "light": "#FAF0F0",
                "main": "#DE1B1B",
              },
              "information": Object {
                "light": "#EAF5FA",
                "main": "#229BD8",
              },
              "success": Object {
                "light": "#F4FCE3",
                "main": "#74B816",
              },
              "warning": Object {
                "light": "#FFF9DB",
                "main": "#F59F00",
              },
            },
            "grey": Object {
              "10": "#FAFAFA",
              "100": "#212121",
              "20": "#F5F5F5",
              "30": "#EEEEEE",
              "40": "#E0E0E0",
              "50": "#BDBDBD",
              "60": "#9E9E9E",
              "70": "#757575",
              "80": "#616161",
              "90": "#424242",
            },
            "text": Object {
              "primary": Object {
                "dark": "#F55151",
                "light": "#900E0E",
              },
              "secondary": Object {
                "dark": "#AD73E8",
                "light": "#632B9C",
              },
              "tertiary": Object {
                "dark": "#AD73E8",
                "light": "#632B9C",
              },
            },
          },
          "pxToRem": [Function],
          "typography": Object {
            "body1": Object {
              "fontSize": "16px",
              "fontWeight": 400,
            },
            "body2": Object {
              "fontSize": "14px",
              "fontWeight": 400,
            },
            "button": Object {
              "fontSize": "14px",
              "fontWeight": 700,
            },
            "caption": Object {
              "fontSize": "12px",
              "fontWeight": 400,
            },
            "fontFamily": "Roboto",
            "fontSize": 14,
            "fontSizeInherit": "inherit",
            "fontWeightBold": 700,
            "fontWeightLight": 300,
            "fontWeightMedium": 500,
            "fontWeightRegular": 400,
            "h1": Object {
              "fontSize": "60px",
              "fontWeight": 300,
            },
            "h2": Object {
              "fontSize": "48px",
              "fontWeight": 400,
            },
            "h3": Object {
              "fontSize": "34px",
              "fontWeight": 400,
            },
            "h4": Object {
              "fontSize": "24px",
              "fontWeight": 400,
            },
            "h5": Object {
              "fontSize": "20px",
              "fontWeight": 500,
            },
            "overline1": Object {
              "fontSize": "10px",
              "fontWeight": 400,
            },
            "overline2": Object {
              "fontSize": "10px",
              "fontWeight": 500,
            },
            "subtitle1": Object {
              "fontSize": "16px",
              "fontWeight": 400,
            },
            "subtitle2": Object {
              "fontSize": "14px",
              "fontWeight": 500,
            },
          },
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "_renderid": "rendered-component",
          "input": Object {},
          "meta": Object {
            "color": "primary",
            "disabled": "false",
            "error": null,
            "helper": "",
            "icon": null,
            "icon-align": "right",
            "placeholder": "",
            "type": null,
            "variant": "basic-filled",
          },
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": null,
        "type": [MockFunction],
      },
      "type": [Function],
    },
    Symbol(enzyme.__nodes__): Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Jss(Textfield)
            _renderid="rendered-component"
            input={Object {}}
            meta={
              Object {
                "color": "primary",
                "disabled": "false",
                "error": null,
                "helper": "",
                "icon": null,
                "icon-align": "right",
                "placeholder": "",
                "type": null,
                "variant": "basic-filled",
              }
            }
            onClick={[MockFunction]}
          />,
          "theme": Object {
            "palette": Object {
              "base": Object {
                "primary": Object {
                  "dark": "#BA0D0D",
                  "light": "#F55151",
                  "main": "#DE1B1B",
                },
                "secondary": Object {
                  "dark": "#7D3CBE",
                  "light": "#AD73E8",
                  "main": "#9657D6",
                },
                "tertiary": Object {
                  "dark": "#1D5BBF",
                  "light": "#5691F0",
                  "main": "#3071D9",
                },
              },
              "common": Object {
                "black": "#212121",
                "white": "#FFFFFF",
              },
              "event": Object {
                "disabled": Object {
                  "main": "#BDBDBD",
                },
                "error": Object {
                  "light": "#FAF0F0",
                  "main": "#DE1B1B",
                },
                "information": Object {
                  "light": "#EAF5FA",
                  "main": "#229BD8",
                },
                "success": Object {
                  "light": "#F4FCE3",
                  "main": "#74B816",
                },
                "warning": Object {
                  "light": "#FFF9DB",
                  "main": "#F59F00",
                },
              },
              "grey": Object {
                "10": "#FAFAFA",
                "100": "#212121",
                "20": "#F5F5F5",
                "30": "#EEEEEE",
                "40": "#E0E0E0",
                "50": "#BDBDBD",
                "60": "#9E9E9E",
                "70": "#757575",
                "80": "#616161",
                "90": "#424242",
              },
              "text": Object {
                "primary": Object {
                  "dark": "#F55151",
                  "light": "#900E0E",
                },
                "secondary": Object {
                  "dark": "#AD73E8",
                  "light": "#632B9C",
                },
                "tertiary": Object {
                  "dark": "#AD73E8",
                  "light": "#632B9C",
                },
              },
            },
            "pxToRem": [Function],
            "typography": Object {
              "body1": Object {
                "fontSize": "16px",
                "fontWeight": 400,
              },
              "body2": Object {
                "fontSize": "14px",
                "fontWeight": 400,
              },
              "button": Object {
                "fontSize": "14px",
                "fontWeight": 700,
              },
              "caption": Object {
                "fontSize": "12px",
                "fontWeight": 400,
              },
              "fontFamily": "Roboto",
              "fontSize": 14,
              "fontSizeInherit": "inherit",
              "fontWeightBold": 700,
              "fontWeightLight": 300,
              "fontWeightMedium": 500,
              "fontWeightRegular": 400,
              "h1": Object {
                "fontSize": "60px",
                "fontWeight": 300,
              },
              "h2": Object {
                "fontSize": "48px",
                "fontWeight": 400,
              },
              "h3": Object {
                "fontSize": "34px",
                "fontWeight": 400,
              },
              "h4": Object {
                "fontSize": "24px",
                "fontWeight": 400,
              },
              "h5": Object {
                "fontSize": "20px",
                "fontWeight": 500,
              },
              "overline1": Object {
                "fontSize": "10px",
                "fontWeight": 400,
              },
              "overline2": Object {
                "fontSize": "10px",
                "fontWeight": 500,
              },
              "subtitle1": Object {
                "fontSize": "16px",
                "fontWeight": 400,
              },
              "subtitle2": Object {
                "fontSize": "14px",
                "fontWeight": 500,
              },
            },
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "_renderid": "rendered-component",
            "input": Object {},
            "meta": Object {
              "color": "primary",
              "disabled": "false",
              "error": null,
              "helper": "",
              "icon": null,
              "icon-align": "right",
              "placeholder": "",
              "type": null,
              "variant": "basic-filled",
            },
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": null,
          "type": [MockFunction],
        },
        "type": [Function],
      },
    ],
    Symbol(enzyme.__options__): Object {
      "adapter": ReactSixteenAdapter {
        "options": Object {
          "enableComponentDidUpdateOnSetState": true,
          "legacyContextMode": "parent",
          "lifecycles": Object {
            "componentDidUpdate": Object {
              "onSetState": true,
            },
            "getChildContext": Object {
              "calledByRenderer": false,
            },
            "getDerivedStateFromError": true,
            "getDerivedStateFromProps": Object {
              "hasShouldComponentUpdateBug": false,
            },
            "getSnapshotBeforeUpdate": true,
            "setState": Object {
              "skipsComponentDidUpdateOnNullish": true,
            },
          },
        },
      },
      Symbol(enzyme.__providerValues__): undefined,
    },
    Symbol(enzyme.__providerValues__): Map {},
    Symbol(enzyme.__childContext__): Object {
      "6fc570d6bd61383819d0f9e7407c452d": Object {
        "generateClassName": [Function],
      },
      "b768b78919504fba9de2c03545c5cd3a": Object {},
      "d4bd0baacbc52bbd48bbb9eb24344ecd": SheetsRegistry {
        "registry": Array [],
      },
    },
  },
  Symbol(enzyme.__unrendered__): null,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "_renderid": "rendered-component",
      "input": Object {},
      "meta": Object {
        "color": "primary",
        "disabled": "false",
        "error": null,
        "helper": "",
        "icon": null,
        "icon-align": "right",
        "placeholder": "",
        "type": null,
        "variant": "basic-filled",
      },
      "onClick": [MockFunction],
    },
    "ref": null,
    "rendered": null,
    "type": [MockFunction],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "_renderid": "rendered-component",
        "input": Object {},
        "meta": Object {
          "color": "primary",
          "disabled": "false",
          "error": null,
          "helper": "",
          "icon": null,
          "icon-align": "right",
          "placeholder": "",
          "type": null,
          "variant": "basic-filled",
        },
        "onClick": [MockFunction],
      },
      "ref": null,
      "rendered": null,
      "type": [MockFunction],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__rootNodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Jss(Textfield)
          _renderid="rendered-component"
          input={Object {}}
          meta={
            Object {
              "color": "primary",
              "disabled": "false",
              "error": null,
              "helper": "",
              "icon": null,
              "icon-align": "right",
              "placeholder": "",
              "type": null,
              "variant": "basic-filled",
            }
          }
          onClick={[MockFunction]}
        />,
        "theme": Object {
          "palette": Object {
            "base": Object {
              "primary": Object {
                "dark": "#BA0D0D",
                "light": "#F55151",
                "main": "#DE1B1B",
              },
              "secondary": Object {
                "dark": "#7D3CBE",
                "light": "#AD73E8",
                "main": "#9657D6",
              },
              "tertiary": Object {
                "dark": "#1D5BBF",
                "light": "#5691F0",
                "main": "#3071D9",
              },
            },
            "common": Object {
              "black": "#212121",
              "white": "#FFFFFF",
            },
            "event": Object {
              "disabled": Object {
                "main": "#BDBDBD",
              },
              "error": Object {
                "light": "#FAF0F0",
                "main": "#DE1B1B",
              },
              "information": Object {
                "light": "#EAF5FA",
                "main": "#229BD8",
              },
              "success": Object {
                "light": "#F4FCE3",
                "main": "#74B816",
              },
              "warning": Object {
                "light": "#FFF9DB",
                "main": "#F59F00",
              },
            },
            "grey": Object {
              "10": "#FAFAFA",
              "100": "#212121",
              "20": "#F5F5F5",
              "30": "#EEEEEE",
              "40": "#E0E0E0",
              "50": "#BDBDBD",
              "60": "#9E9E9E",
              "70": "#757575",
              "80": "#616161",
              "90": "#424242",
            },
            "text": Object {
              "primary": Object {
                "dark": "#F55151",
                "light": "#900E0E",
              },
              "secondary": Object {
                "dark": "#AD73E8",
                "light": "#632B9C",
              },
              "tertiary": Object {
                "dark": "#AD73E8",
                "light": "#632B9C",
              },
            },
          },
          "pxToRem": [Function],
          "typography": Object {
            "body1": Object {
              "fontSize": "16px",
              "fontWeight": 400,
            },
            "body2": Object {
              "fontSize": "14px",
              "fontWeight": 400,
            },
            "button": Object {
              "fontSize": "14px",
              "fontWeight": 700,
            },
            "caption": Object {
              "fontSize": "12px",
              "fontWeight": 400,
            },
            "fontFamily": "Roboto",
            "fontSize": 14,
            "fontSizeInherit": "inherit",
            "fontWeightBold": 700,
            "fontWeightLight": 300,
            "fontWeightMedium": 500,
            "fontWeightRegular": 400,
            "h1": Object {
              "fontSize": "60px",
              "fontWeight": 300,
            },
            "h2": Object {
              "fontSize": "48px",
              "fontWeight": 400,
            },
            "h3": Object {
              "fontSize": "34px",
              "fontWeight": 400,
            },
            "h4": Object {
              "fontSize": "24px",
              "fontWeight": 400,
            },
            "h5": Object {
              "fontSize": "20px",
              "fontWeight": 500,
            },
            "overline1": Object {
              "fontSize": "10px",
              "fontWeight": 400,
            },
            "overline2": Object {
              "fontSize": "10px",
              "fontWeight": 500,
            },
            "subtitle1": Object {
              "fontSize": "16px",
              "fontWeight": 400,
            },
            "subtitle2": Object {
              "fontSize": "14px",
              "fontWeight": 500,
            },
          },
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "_renderid": "rendered-component",
          "input": Object {},
          "meta": Object {
            "color": "primary",
            "disabled": "false",
            "error": null,
            "helper": "",
            "icon": null,
            "icon-align": "right",
            "placeholder": "",
            "type": null,
            "variant": "basic-filled",
          },
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": null,
        "type": [MockFunction],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__providerValues__): null,
}
`;
